src/Esprima/Utils/JsonWriter.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Esprima.Utils
+{
+    public abstract class JsonWriter
```

Comment:
JSON makes sense here because JSON is the _model_ for the API. The JSON that you're talking about, and which most people understand to be [RFC 4627](https://tools.ietf.org/html/rfc4627)/[7159](https://tools.ietf.org/html/rfc7159), is [grammar-wise](https://tools.ietf.org/html/rfc7159#section-2) or syntactically known as _JSON text_:

> # 2.  JSON Grammar
>
> A JSON text is a sequence of tokens.  The set of tokens includes six structural characters, strings, numbers, and three literal names.
>
> A JSON text is a serialized object or array.
>
>     JSON-text = object / array

So the implementation of `JsonWriter` that emits JSON text is called `JsonTextWriter`.